--- a/qtbase/src/corelib/global/qglobal.h
+++ b/qtbase/src/corelib/global/qglobal.h
@@ -108,6 +108,7 @@
 #  include <type_traits>
 #  include <cstddef>
 #  include <utility>
+#  include <limits>
 #endif
 
 #ifndef __ASSEMBLER__
--- a/qtbase/src/corelib/tools/qbytearraymatcher.h
+++ b/qtbase/src/corelib/tools/qbytearraymatcher.h
@@ -40,6 +40,8 @@
 #ifndef QBYTEARRAYMATCHER_H
 #define QBYTEARRAYMATCHER_H
 
+#include <limits>
+
 #include <QtCore/qbytearray.h>
 
 QT_BEGIN_NAMESPACE
