--- a/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp
+++ b/qtbase/src/plugins/platforms/windows/qwindowsmousehandler.cpp
@@ -60,7 +60,7 @@
  * are present in the Windows SDK's, but not in older MSVC Express
  * versions. */
 
-#if defined(Q_CC_MINGW) || !defined(TOUCHEVENTF_MOVE)
+#if (!defined(__MINGW32__) && defined(Q_CC_MINGW)) || !defined(TOUCHEVENTF_MOVE)
 
 typedef struct tagTOUCHINPUT {
     LONG x;
